mysql_fdw.c

```diff
@@ -942,11 +950,15 @@ mysqlGetForeignPlan(PlannerInfo *root,RelOptInfo *baserel, Oid foreigntableid, F
 	 * field of the finished plan node; we can't keep them in private state
 	 * because then they wouldn't be subject to later planner processing.
 	 */
-	return make_foreignscan(tlist,
-							local_exprs,
-							scan_relid,
-							params_list,
-							fdw_private);
+	return make_foreignscan(tlist
```

Comment:
On Tue, Jul 7, 2015 at 9:33 PM, jjthiessen <notifications@github.com> wrote:

> In mysql_fdw.c
> <https://github.com/EnterpriseDB/mysql_fdw/pull/62#discussion_r34060458>:
>
> > @@ -942,11 +950,15 @@ mysqlGetForeignPlan(PlannerInfo *root,RelOptInfo *baserel, Oid foreigntableid, F
> >  	 * field of the finished plan node; we can't keep them in private state
> >  	 * because then they wouldn't be subject to later planner processing.
> >  	 */
> > -	return make_foreignscan(tlist,
> > -							local_exprs,
> > -							scan_relid,
> > -							params_list,
> > -							fdw_private);
> > +	return make_foreignscan(tlist
>
> I have yet to be able to reproduce the crash that you're reporting (I've
> admittedly done very limited testing). Could you please provide the SQL
> that you used when you encountered the crash (on both the MySQL and
> PostgreSQL sides)? Could you also please provide the PostgreSQL commit that
> you were on? Any other details about your build environment would be
> appreciated.
>

I can reproduce the crash consistently using the following steps :

1- Compiled and installed mysql fdw by pulling the latest code from github
2- Applied the attached patch for import foreign schema functionality and
the import schema was done fine
3- Tried to create another mysql fdw foreign table and did select from it
and it crashed.
4- Crash is not reproducible without the patch

I am using PG 9.5 alpha and mysql 5.6.19.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/EnterpriseDB/mysql_fdw/pull/62/files#r34060458>.
>



-- 
Ahsan Hadi
Snr Director Product Development
EnterpriseDB Corporation
The Enterprise Postgres Company

Phone: +92-51-8358874
Mobile: +92-333-5162114

Website: www.enterprisedb.com
EnterpriseDB Blog: http://blogs.enterprisedb.com/
Follow us on Twitter: http://www.twitter.com/enterprisedb

This e-mail message (and any attachment) is intended for the use of the
individual or entity to whom it is addressed. This message contains
information from EnterpriseDB Corporation that may be privileged,
confidential, or exempt from disclosure under applicable law. If you are
not the intended recipient or authorized to receive this for the intended
recipient, any use, dissemination, distribution, retention, archiving, or
copying of this communication is strictly prohibited. If you have received
this e-mail in error, please notify the sender immediately by reply e-mail
and delete this message.